src/Analysis/Ast/Impl/Diagnostics/DiagnosticsEntry.cs

```diff
@@ -51,6 +53,26 @@ public sealed class DiagnosticsEntry {
         /// </summary>
         public DiagnosticSource Source { get; }
 
+        public bool ShouldReport(IPythonModule module) {
+            // module should always be a user written python module
+            if (module.ModuleType != ModuleType.User || !(module is PythonModule pythonModule)) {
```

Comment:
`!(module is PythonModule pythonModule)` shouldn't be needed. Do you have a case when `module.ModuleType == ModuleType.User` but `module` isn't a `PythonModule`?

Reply:
I need to make the conversion between IPythonModule to PythonModule and I didn't feel comfortable just doing it without a matching check.